source/loader/ur_ldrddi.cpp

```diff
@@ -102,6 +102,13 @@ __urdlllocal ur_result_t UR_APICALL urAdapterRelease(
     // forward to device-platform
     result = pfnAdapterRelease(hAdapter);
 
+    if (UR_RESULT_SUCCESS != result) {
+        return result;
+    }
+
+    // release loader handle
+    ur_adapter_factory.release(hAdapter);
```

Comment:
eh.. it occurred to me as a result of adding an equivalent [retain] tag in #1759 that this isn't reference counted, it's just unconditionally releasing on the first `Release` call that comes in, might need to add a reference counting mechanism to the factory objects